test/src/main/java/org/apache/accumulo/test/OfflineTableIT.java

```diff
@@ -104,4 +127,100 @@ public void testSplitOffline() throws Exception {
     }
   }
 
+  @Test
+  public void testEcompWaitForOffline() throws Exception {
+    final var ctx = getCluster().getServerContext();
+
+    try (AccumuloClient client = Accumulo.newClient().from(getClientProps()).build()) {
+      String tableName = getUniqueNames(1)[0];
+      ScanServerIT.createTableAndIngest(client, tableName, null, 10, 10, "colf");
+      TableId tableId = TableId.of(client.tableOperations().tableIdMap().get(tableName));
+      final var tabletMeta = ctx.getAmple().readTablet(new KeyExtent(tableId, null, null));
+
+      final var ecid = ExternalCompactionId.generate(UUID.randomUUID());
+
+      // Insert a fake External compaction which should prevent the wait for offline
+      // from returning
+      try (var mutator = ctx.getAmple().mutateTablets()) {
+        var tabletDir =
+            tabletMeta.getFiles().stream().findFirst().orElseThrow().getPath().getParent();
+        var tmpFile = new Path(tabletDir, "C1234.rf_tmp");
+        var cm = new CompactionMetadata(tabletMeta.getFiles(), ReferencedTabletFile.of(tmpFile),
+            "localhost:16789", CompactionKind.SYSTEM, (short) 10, CompactorGroupId.of(GROUP1),
+            false, null);
+        mutator.mutateTablet(tabletMeta.getExtent()).putExternalCompaction(ecid, cm).mutate();
+      }
+
+      // test the ecomp prevents the wait for the offline() table operation from finishing
+      // until the ecomp is deleted
+      testWaitForOffline(ctx, client, tableId, tableName, mutator -> mutator
+          .mutateTablet(tabletMeta.getExtent()).deleteExternalCompaction(ecid).mutate());
+    }
+  }
+
+  @Test
+  public void testOpidWaitForOffline() throws Exception {
+    final var ctx = getCluster().getServerContext();
+
+    try (AccumuloClient client = Accumulo.newClient().from(getClientProps()).build()) {
+      String tableName = getUniqueNames(1)[0];
+      ScanServerIT.createTableAndIngest(client, tableName, null, 10, 10, "colf");
+      TableId tableId = TableId.of(client.tableOperations().tableIdMap().get(tableName));
+      final var tabletMeta = ctx.getAmple().readTablet(new KeyExtent(tableId, null, null));
+
+      // Insert a fake opid to prevent going offline
+      try (var mutator = ctx.getAmple().mutateTablets()) {
+        mutator.mutateTablet(tabletMeta.getExtent())
+            .putOperation(TabletOperationId.from(TabletOperationType.SPLITTING,
+                FateId.from(FateInstanceType.META, UUID.randomUUID())))
+            .mutate();
+      }
+
+      // test the opid prevents the wait for the offline() table operation from finishing
+      // until the opid is deleted
+      testWaitForOffline(ctx, client, tableId, tableName,
+          mutator -> mutator.mutateTablet(tabletMeta.getExtent()).deleteOperation().mutate());
+    }
+  }
+
+  private void testWaitForOffline(ServerContext ctx, AccumuloClient client, TableId tableId,
```

Comment:
This is not a comment about a change for this PR, just posting for discussion.  This new method could possibly be used to test existing functionality of waiting for locations.  However that could be tricky as setting fake locations will probably trigger the manager to take actions as it will consider them dead tservers.